File: python/progressive-average.py
sum = 0

for i in range(10):
    inputValue = int(input("Type a number: "))
    sum += inputValue
    if i >= 1:
        print("The current sum is: "  + str(sum) + " | The current index is: " + str(i))
        average = sum / (i + 1) if i == 1 else sum / i
        print("The average is: " + str(average))

File: c/progressive-average.c
#include <stdio.h>

int i;
float sum;
float average;

void main() {
  for (i = 0; i < 10; i++) {
    printf("Type a number: ");
    int number;
    scanf("%d", &number);

    sum += number;

    if (i >= 1) {
      printf("The current sum is: %.1f, The current index is: %d\n", sum, i);
      average = (i == 1) ? sum / 2 : sum / i;
      printf("\nThe average is: %.1f\n", average);
    }
  }
}